lib/DiscoveryStreamFeed.jsm

```diff
@@ -622,6 +670,11 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
       case at.DISCOVERY_STREAM_OPT_OUT:
         this.store.dispatch(ac.SetPref(PREF_OPT_OUT, true));
         break;
+      case at.DISCOVERY_STREAM_IMPRESSION_STATS:
+        if (action.data.tiles && action.data.tiles[0] && action.data.tiles[0].id) {
+          this.recordTopRecImpressions(action.data.tiles[0].id);
```

Comment:
Forgot to update tests based on this new API. Fixing with https://github.com/mozilla/activity-stream/pull/4799